core/domain/stats_jobs_one_off.py

```diff
@@ -1257,6 +1257,14 @@ def map(exp):
             stats_models.ExplorationStatsModel.get_multi_stats_models(
                 [exp_domain.ExpVersionReference(exp.id, version)
                  for version in python_utils.RANGE(1, exp.version + 1)]))
+
+        if exp.deleted:
+            for model in all_models:
+                if model:
+                    model.delete()
+            yield ('Deleted all stats', exp.id)
```

Comment:
Maybe include number of items deleted in the  message.

Reply:
Done.